gamemodes/terrortown/entities/weapons/weapon_zm_carry.lua

```diff
@@ -58,88 +58,90 @@ SWEP.AllowDelete = false
-	end, cvPropThrow:GetName())
+-- constants
+SWEP.pickupRangeRagdoll = 75
+SWEP.pickupRangeMisc = 100
+SWEP.moveForce = 6000
+SWEP.carryDistance = 70
+SWEP.standCheckFrequency = 0.1
+SWEP.entDiffDistanceThreshold = 40000
+SWEP.entDiffCheckFrequency = 1
+SWEP.pinBoneAttachFactor = 0.9
+SWEP.pinBoneLengthFactor = 0.1
+SWEP.pinRopeWidth = 1
+SWEP.pinMaterial = "cable/rope"
+SWEP.pinnedHealth = 999999
+SWEP.constraintType = "Rope"
+SWEP.dropWakeForce = 500
+SWEP.dropAngleThreshold = 0.75
+SWEP.carryMinDamping = 0
+SWEP.carryMaxDamping = 1000
+SWEP.carryMass = 200
+SWEP.carryHealth = 999
+SWEP.carryPushDelay = 0.03
+SWEP.carryPickupDelay = 0.5
+SWEP.carryPickupRagdollDelay = 0.8
+SWEP.carryPickupReleaseDelay = 0.3
+SWEP.moveObjectLightObjectMass = 50
+SWEP.moveObjectLightObjectAdditionalMass = 0.5
+SWEP.moveObjectLightObjectMassScale = 0.02
+SWEP.moveObjectRagdollForceMultiplier = 2
+SWEP.moveObjectRemapSpeedInMin = 0
+SWEP.moveObjectRemapSpeedInMax = 125
+SWEP.moveObjectRemapSpeedInMin = 1
+SWEP.pinRagRange = 90
+
+local flags = {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED}
 
-	cvars.AddChangeCallback(cvAllowRagPin:GetName(), function(name, _, new)
-		SetGlobalBool(name, tonumber(new) == 1)
-	end, cvAllowRagPin:GetName())
+---
+-- @realm shared
+local cvAllowRagCarry = CreateConVar("ttt_ragdoll_carrying", "1", flags)
 
-	cvars.AddChangeCallback(cvAllowRagPinInno:GetName(), function(name, _, new)
-		SetGlobalBool(name, tonumber(new) == 1)
-	end, cvAllowRagPinInno:GetName())
+---
+-- @realm server
+local cvPropForce = CreateConVar("ttt_prop_carrying_force", "60000", flags)
+---
+-- @realm shared
+local cvPropThrow = CreateConVar("ttt_prop_throwing", "1", flags)
 
-	cvars.AddChangeCallback(cvAllowWepCarry:GetName(), function(name, _, new)
-		SetGlobalBool(name, tonumber(new) == 1)
-	end, cvAllowWepCarry:GetName())
+---
+-- @note Allowing weapon pickups can allow players to cause a crash in the physics
+-- system (ie. not fixable). Tuning the range seems to make this more
+-- difficult. Not sure why. It's that kind of crash.
+-- @realm server
+local cvAllowWepCarry = CreateConVar("ttt_weapon_carrying", "0", flags)
 
-	cvars.AddChangeCallback(cvWepCarryRange:GetName(), function(name, _, new)
-		SetGlobalInt(name, tonumber(new))
-	end, cvWepCarryRange:GetName())
+---
+-- @note Allowing weapon pickups can allow players to cause a crash in the physics
+-- system (ie. not fixable). Tuning the range seems to make this more
+-- difficult. Not sure why. It's that kind of crash.
+-- @realm server
+local cvWepCarryRange = CreateConVar("ttt_weapon_carrying_range", "50", flags)
+
+CARRY_TYPE_NONE = 0
+CARRY_TYPE_PROP = 1
+CARRY_TYPE_WEAPON = 2
+CARRY_TYPE_RAGDOLL = 3
+
+local function DetermineCarryType(ent)
+	if IsValid(ent) then
+		if ent.IsWeapon and ent:IsWeapon() then
+			return CARRY_TYPE_WEAPON
+		elseif ent:GetClass() == "prop_ragdoll" then
+			return CARRY_TYPE_RAGDOLL
+		else
+			return CARRY_TYPE_PROP
+		end
+	else
+		return CARRY_TYPE_NONE
+	end
 end
```

Comment:
If this all just derives from the entity then i would rather not store this in another variable and use the function instead. 
This way we have less state complexity

Reply:
what does that mean for the code

Reply:
https://github.com/TTT-2/TTT2/pull/1253#discussion_r1435871993
i think this was the reasoning behind this change, correct?

Reply:
correct

Reply:
The linked comment has nothing to do with this or GH is fucking with me.

Anyways this would mean to remove CarryType as a variable and just call the DetermineCarryType method where necessary

Reply:
The NetworkVar for CarryType still is necessary because the client doesn't always know about the entity afaict, it was largely the problem I had with making the tooltip reactive to what the client was looking at. Some server-sided entities don't exist on the client and it can be confusing.

Reply:
i went back and changed it anyway, it probably works